dz6.py

```diff
@@ -1,32 +1,32 @@
 """Compiling a new list using comprehension."""
 
-from itertools import zip_longest
-
 # lists № 1
+# l1 = [2, 4, 6, 8, 10]
+# l2 = [1, 2, 3]
+
+# lists № 2
 l1 = [1, 3, 5, 7]
 l2 = [1, 4, 5]
 
-# lists № 2
+# lists № 3
 # l1 = [1, 4, 5]
 # l2 = [1, 3, 5, 7]
 
-# lists № 3
-# l1 = [2, 4, 6, 8, 10]
-# l2 = [1, 2, 3]
+# Finding common indexes from both lists
+index_l1 = set(range(len(l1)))
+index_l2 = set(range(len(l2)))
+index_union = index_l1.union(index_l2)
 
-# Get the maximum length between the two lists
-max_length = max(len(l1), len(l2))
 
-# Using zip_longest to pair up elements and
-# filling the missing elements digit 0
-var1 = zip_longest(l1, l2, fillvalue=0)
+# Creating a dictionary of values
+result_dict = {i: (l1[i] if i < len(l1) else 0,
```

Comment:
0 better to place as variable for example `default_value=0` and then use it inside code
This is best practice -> avoid hardcodes